playbooks/kubevirt.yml

```diff
@@ -0,0 +1,260 @@
+    - epel_repositories
+  tasks:
+    - name: disable swap
+      command: swapoff -a
+
+    - name: deconfigure swap
+      mount:
+        src: /swapfile
+        fstype: swap
+        path: none
+        state: absent
+
+    - name: modprobe br_netfilter
+      modprobe:
+        name: br_netfilter
+
+    - name: configure bridge iptables
+      sysctl:
+        name: "{{ item }}"
+        value: 1
+        sysctl_file: /etc/sysctl.d/k8s.conf
+      with_items:
+        - net.bridge.bridge-nf-call-ip6tables
+        - net.bridge.bridge-nf-call-iptables
+
+    - name: install needed network manager libs
+      yum:
+        name:
+        - NetworkManager-glib
+        - NetworkManager
+
+    - name: Configure bridge
+      nmcli:
+        state: present
+        type: bridge
+        conn_name: br0
+
+    - name: install docker
+      yum:
+        name: docker
+
+    - name: enable docker
+      service:
+        name: docker
+        enabled: true
+        state: started
+
+    - name: k8s repo
+      yum_repository:
+        name: kubernetes
+        description: Kubernetes
+        baseurl: https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64
+        enabled: true
+        gpgcheck: true
+        repo_gpgcheck: true
+        gpgkey: https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+        exclude: kube*
+
+    - name: install kubelet kubeadm kubectl
+      yum:
+        name:
+          - kubelet-{{ kubernetes_version }}
+          - kubeadm-{{ kubernetes_version }}
+          - kubectl-{{ kubernetes_version }}
+        disable_excludes: kubernetes
+
+    - name: create /etc/systemd/system/kubelet.service.d/
+      file:
+        path: /etc/systemd/system/kubelet.service.d/
+        state: directory
+
+    - name: enable kubelet accounting
+      copy:
+        dest: /etc/systemd/system/kubelet.service.d/11-cgroups.conf
+        content: |
+          [Service]
+          CPUAccounting=true
+          MemoryAccounting=true
+
+    - name: reload systemd
+      systemd:
+        daemon_reload: yes
+
+    - name: enable kubelet
+      service:
+        name: kubelet
+        enabled: true
+        state: started
+
+    - name: init cluster
+      command: kubeadm init --pod-network-cidr={{ pod_network }} --apiserver-advertise-address={{ ansible_eth0['ipv4']['address'] }}
+      args:
+        creates: /etc/kubernetes/admin.conf
+
+    - name: untaint master
+      command: kubectl taint nodes --all node-role.kubernetes.io/master-
+      register: untaint_master
+      failed_when: false
+      changed_when: untaint_master.rc == 0
+      environment:
+        KUBECONFIG: /etc/kubernetes/admin.conf
+
+    - name: deploy network namespace
+      command: kubectl apply -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/{{ network_operator }}/namespace.yaml
+      args:
+        creates: /etc/cni/net.d/multus.d/multus.kubeconfig
```

Comment:
not a master in Ansible, but how is namespace supposed to create multus configuration?

Reply:
AFAIU this isn't required since the cluster network operator should deploy multus and set its configuration

Reply:
It will deploy multus only if you ask it to do so, via its CR https://github.com/kubevirt/cluster-network-addons-operator#multus. I don't get why it is being mentioned in this step, it should be only done when the CR is created